SConstruct

```diff
@@ -334,13 +413,29 @@ elif env['platform'] == 'android':
     }
     arch_info = arch_info_table[env['android_arch']]
 
+    cmd = ""
+    if host_platform == "windows":
+        cmd = ".cmd"
+
     # Setup tools
-    env['CC'] = toolchain + "/bin/clang"
-    env['CXX'] = toolchain + "/bin/clang++"
-    env['AR'] = toolchain + "/bin/" + arch_info['tool_path'] + "-ar"
+    if standalone:
+        if os.path.exists(toolchain + "/bin/" + arch_info["tool_path"] + "-clang" + cmd):
+            cc = "clang"
+            cxx = "clang++"
+            # I'd use NDK version for this, but that isn't defined when the toolchain is manually set.
+        else:
+            cc = "gcc"
+            cxx = "g++"
+        env['CC'] = '"' + toolchain + "/bin/" + arch_info["tool_path"] + '-' + cc + cmd + '"'
+        env['CXX'] = '"' + toolchain + "/bin/" + arch_info["tool_path"] + '-' + cxx + cmd + '"'
+    else:
+        env['CC'] = '"' + toolchain + "/bin/clang" + '"'
+        env['CXX'] = '"' + toolchain + "/bin/clang++" + '"'
+    env['AR'] = '"' + toolchain + "/bin/" + arch_info['tool_path'] + "-ar" + '"'
```

Comment:
Incremental builds won't work without:

```
    env["RANLIB"] = '"' + toolchain + "/bin/" + arch_info['tool_path'] + "-ranlib" + '"'
```

Reply:
Sorry for not responding. Added RANLIB and bumped minimum NDK version to 17.